tests/unit_tests/template.py

```diff
@@ -0,0 +1,614 @@
+                            "growth_rate": "0.0", 
+                            "initial_demand": "2", 
+                            "prototypes": {"val": "source"}, 
+                            "steps": "1", 
+                            "supply_commod": "fuel"
+                           }
+                          }, 
+                          "initialfacilitylist":{
+                              "entry":{"number":1,
+                                       "prototype":"source"}
+                          }, 
+                          "name": "source_inst"
+                         }, 
+                         "name": "SingleRegion"
+                         }
+                        }
+                        )
+
+
+def test_a2_init_demand_with_init_facilities():
+    # tests if NOInst deploys a source given initial demand and no initial facilities
+    output_file = 'init_demand_init_fac.sqlite'
+    input_file = output_file.replace('.sqlite','.json')
+    with open(input_file, 'w') as f:
+        json.dump(INIT_DEMAND_WITH_INIT_FACILITIES, f)
+    s = subprocess.check_output(['cyclus', '-o', output_file, input_file],
+                                universal_newlines=True, env=ENV)
+    # check if ran successfully
+    assert("Cyclus run successful!" in s)
+
+    # getting the sqlite file
+    cur = get_cursor(output_file)
+    # check if 10 source facilities were deployed by NOInst
+    source = cur.execute("SELECT count(*) FROM agententry WHERE Prototype = 'source'"
+                         " AND EnterTime = 1").fetchone()
+    assert(source[0] == 1)
+
+
+# Test A_3 
+INCREASING_DEMAND = copy.deepcopy(TEMPLATE)
+INCREASING_DEMAND['simulation'].update(
+   {"region": {
+   "config": {"NullRegion": "\n      "}, 
+   "institution": {
+    "config": {
+     "NOInst": {
+      "calc_method": "arma", 
+      "demand_commod": "POWER", 
+      "demand_std_dev": "0.0", 
+      "growth_rate": "1.0", 
+      "initial_demand": "1", 
+      "prototypes": {"val": "source"}, 
+      "steps": "1", 
+      "supply_commod": "fuel"
+     }
+    }, 
+    "name": "source_inst"
+   }, 
+   "name": "SingleRegion"
+  }
+  }
+    )
+
+def test_a3_increasing_demand():
+    # tests if NOInst deploys a source according to increasing demand
+    output_file = 'increasing_demand.sqlite'
+    input_file = output_file.replace('.sqlite','.json')
+    with open(input_file, 'w') as f:
+        json.dump(INCREASING_DEMAND, f)
+    s = subprocess.check_output(['cyclus', '-o', output_file, input_file],
+                                universal_newlines=True, env=ENV)
+    # check if ran successfully
+    assert("Cyclus run successful!" in s)
+    # getting the sqlite file
+    cur = get_cursor(output_file)
+    # check if 1 source facility was deployed by NOInst in timestep 1
+    source = cur.execute("SELECT count(*) FROM agententry WHERE Prototype = 'source'"
+                         " AND EnterTime = 1").fetchone()
+    assert(source[0] == 2)
```

Comment:
I think in general this type of tests should be that 'at least' one is deployed.  I'm okay with setting a hard equal but I'm more comfortable with 1 or more being deployed. 

Also this is checking that at least 2 are deployed on time step one. Is this correct?

Reply:
Yes that is correct. We can implement the test to be 1 or more being deployed instead of just at time step 1.

Reply:
So should we have one that passes if one or more deploys and another one to test if it over-supplies?

Reply:
I would say if the supply is more than one `prod_rate`, or in other words, one unit capacity of the facility.